platform/mellanox/mlnx-fw-upgrade.j2

```diff
@@ -4,39 +4,65 @@ declare -r SCRIPT_NAME="$(basename "$0")"
 declare -r SCRIPT_PATH="$(readlink -f "$0")"
 declare -r SCRIPT_DIR="$(dirname "$SCRIPT_PATH")"
 
+declare -r YES_PARAM="yes"
+declare -r NO_PARAM="no"
+
+declare -r VERBOSE_ERROR="1"
+declare -r VERBOSE_WARNING="2"
+declare -r VERBOSE_NOTICE="3"
+declare -r VERBOSE_INFO="4"
+
+declare -r VERBOSE_MAX="${VERBOSE_INFO}"
+declare -r VERBOSE_MIN="${VERBOSE_ERROR}"
+
 declare -r EXIT_SUCCESS="0"
-declare -r EXIT_ERROR="1"
+declare -r EXIT_FAILURE="1"
 
 declare -r QUERY_CMD="mlxfwmanager --query"
 declare -r BURN_CMD="mlxfwmanager -u -f -y"
 
-declare -r FW_FILE="/etc/mlnx/fw-SPC.mfa"
-declare -r QUERY_FILE="/tmp/mlnxfwmanager-query.txt"
+declare -r QUERY_FILE="/tmp/mlxfwmanager-query.log"
 
-declare -r FW_REQUIRED="{{ MLNX_FW_VERSION }}"
+declare -r SPC1_ASIC="spc1"
+declare -r SPC2_ASIC="spc2"
+declare -r UNKN_ASIC="unknown"
 
-IMAGE_UPGRADE="no"
+declare -rA FW_FILE_MAP=( \
+    [$SPC1_ASIC]="/etc/mlnx/fw-SPC.mfa" \
+    [$SPC2_ASIC]="/etc/mlnx/fw-SPC2.mfa" \
+)
+declare -rA FW_REQUIRED_MAP=( \
+    [$SPC1_ASIC]="{{ MLNX_SPC_FW_VERSION }}" \
+    [$SPC2_ASIC]="{{ MLNX_SPC2_FW_VERSION }}" \
+)
+
+IMAGE_UPGRADE="${NO_PARAM}"
+VERBOSE_LEVEL="${VERBOSE_MIN}"
 
 function PrintHelp() {
     echo
     echo "Usage: ./${SCRIPT_NAME} [OPTIONS]"
     echo
     echo "OPTIONS:"
-    echo "  -u, --upgrade  Upgrade MLNX ASIC firmware using next boot image (useful after SONiC-To-SONiC update)"
+    echo "  -u, --upgrade  Upgrade ASIC firmware using next boot image (useful after SONiC-To-SONiC update)"
```

Comment:
as this is MLNX ASIC, any reason why taking MLNX out of it? In any case suggest to use Mellanox and not MLNXS